sdk/scheduler/src/main/java/com/mesosphere/sdk/state/StateStoreCache.java

```diff
@@ -216,7 +222,13 @@ public void clearTask(String taskName) throws StateStoreException {
     public Optional<TaskInfo> fetchTask(String taskName) throws StateStoreException {
         RLOCK.lock();
         try {
-            return Optional.ofNullable(nameToTask.get(taskName));
+            if (nameToTask.containsKey(taskName)) {
```

Comment:
Because we don't allow storing `null` tasks in `nameToTask` map. I think it's fine to drop the `containsKey` check.